caffe2/python/operator_test/conv_test.py

```diff
@@ -315,17 +373,59 @@ def test_3d_convolution_cudnn_nchw(self, batch_size, stride, size, kernel,
         filter_dims = [output_channels, input_channels]
         filter_dims.extend([kernel] * n)
         X = np.random.rand(*input_dims).astype(np.float32) - 0.5
-        w = np.random.rand(*filter_dims).astype(np.float32) - 0.5
+        W = np.random.rand(*filter_dims).astype(np.float32) - 0.5
         b = np.random.rand(output_channels).astype(np.float32) - 0.5
 
-        inputs = [X, w, b] if use_bias else [X, w]
+        inputs = [X, W, b] if use_bias else [X, W]
 
         if size + pad + pad < dkernel or size + pad + pad < dkernel:
-            with self.assertRaises(RuntimeError):
+            with self.assertraises(runtimeerror):
                 self.assertDeviceChecks(dc, op, inputs, [0])
             return
 
         self.assertDeviceChecks(dc, op, inputs, [0])
+
+    @given(batch_size=st.integers(1, 2),
+           stride=st.integers(1, 2),
+           size=st.integers(3, 5),
+           kernel=st.integers(1, 2),
+           dilation=st.integers(1, 2),
+           pad=st.integers(0, 2),
+           use_bias=st.booleans(),
+           **hu.gcs_gpu_only)
```

Comment:
Ok! can you please annotate the test to run only when GPU device is available ?
`@unittest.skipIf(not workspace.has_gpu_support, "No gpu support.")`
Otherwise it will crash on CPU only machines.